From d43eb505ca55acdda3d4ae7c4582d2ee9105dce8 Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Wed, 8 Oct 2025 01:05:58 +0300
Subject: [PATCH 360/553] tst_QTextEdit: port away from QPair

Pick-to: 6.8 6.5
Task-number: QTBUG-115841
Change-Id: I3a749cab756f77641d698e5413561a63e5a425c9
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit f03329296bb8d323a2df5ecfb19decd1ae7b1647)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../widgets/qtextedit/tst_qtextedit.cpp       | 70 +++++++++++--------
 1 file changed, 39 insertions(+), 31 deletions(-)

diff --git a/tests/auto/widgets/widgets/qtextedit/tst_qtextedit.cpp b/tests/auto/widgets/widgets/qtextedit/tst_qtextedit.cpp
index 36dadf5f213..23437a41c56 100644
--- a/tests/auto/widgets/widgets/qtextedit/tst_qtextedit.cpp
+++ b/tests/auto/widgets/widgets/qtextedit/tst_qtextedit.cpp
@@ -38,7 +38,7 @@
 Q_LOGGING_CATEGORY(lcTests, "qt.widgets.tests")
 
 //Used in copyAvailable
-typedef QPair<Qt::Key, Qt::KeyboardModifier> keyPairType;
+using keyPairType = std::pair<Qt::Key, Qt::KeyboardModifier>;
 typedef QList<keyPairType> pairListType;
 Q_DECLARE_METATYPE(keyPairType);
 
@@ -1281,71 +1281,79 @@ void tst_QTextEdit::copyAvailable_data()
     QTest::addColumn<QList<bool> >("copyAvailable");
     QTest::addColumn<QString>("function");
 
-    pairListType keystrokes;
     QList<bool> copyAvailable;
-
-    keystrokes << qMakePair(Qt::Key_B, Qt::NoModifier) <<  qMakePair(Qt::Key_B, Qt::NoModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier);
+    pairListType keystrokes = {{Qt::Key_B, Qt::NoModifier},
+                               {Qt::Key_B, Qt::NoModifier},
+                               {Qt::Key_Left, Qt::ShiftModifier}};
     copyAvailable << true ;
     QTest::newRow(QString("Case1 B,B, <- + shift | signals: true").toLatin1())
         << keystrokes << copyAvailable << QString();
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_T, Qt::NoModifier) << qMakePair(Qt::Key_A, Qt::NoModifier)
-               <<  qMakePair(Qt::Key_A, Qt::NoModifier) << qMakePair(Qt::Key_Left, Qt::ShiftModifier);
+    keystrokes = {{Qt::Key_T, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier}};
     copyAvailable << true << false;
     QTest::newRow(QString("Case2 T,A,A, <- + shift, cut() | signals: true, false").toLatin1())
         << keystrokes << copyAvailable << QString("cut");
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_T, Qt::NoModifier) << qMakePair(Qt::Key_A, Qt::NoModifier)
-               <<  qMakePair(Qt::Key_A, Qt::NoModifier) << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier)  << qMakePair(Qt::Key_Left, Qt::ShiftModifier);
+    keystrokes = {{Qt::Key_T, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier}};
     copyAvailable << true;
     QTest::newRow(QString("Case3 T,A,A, <- + shift, <- + shift, <- + shift, copy() | signals: true").toLatin1())
         << keystrokes << copyAvailable << QString("copy");
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_T, Qt::NoModifier) << qMakePair(Qt::Key_A, Qt::NoModifier)
-               <<  qMakePair(Qt::Key_A, Qt::NoModifier) << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier)  << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_X, Qt::ControlModifier);
+    keystrokes = {{Qt::Key_T, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_X, Qt::ControlModifier}};
     copyAvailable << true << false;
     QTest::newRow(QString("Case4 T,A,A, <- + shift, <- + shift, <- + shift, ctrl + x, paste() | signals: true, false").toLatin1())
         << keystrokes << copyAvailable << QString("paste");
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_B, Qt::NoModifier) <<  qMakePair(Qt::Key_B, Qt::NoModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier) << qMakePair(Qt::Key_Left, Qt::NoModifier);
+    keystrokes = {{Qt::Key_B, Qt::NoModifier},
+                  {Qt::Key_B, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::NoModifier}};
     copyAvailable << true << false;
     QTest::newRow(QString("Case5 B,B, <- + shift, <- | signals: true, false").toLatin1())
         << keystrokes << copyAvailable << QString();
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_B, Qt::NoModifier) <<  qMakePair(Qt::Key_A, Qt::NoModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier) << qMakePair(Qt::Key_Left, Qt::NoModifier)
-               << qMakePair(Qt::Key_Right, Qt::ShiftModifier);
+    keystrokes = {{Qt::Key_B, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::NoModifier},
+                  {Qt::Key_Right, Qt::ShiftModifier}};
     copyAvailable << true << false << true << false;
     QTest::newRow(QString("Case6 B,A, <- + shift, ->, <- + shift | signals: true, false, true, false").toLatin1())
         << keystrokes << copyAvailable << QString("cut");
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_T, Qt::NoModifier) << qMakePair(Qt::Key_A, Qt::NoModifier)
-               <<  qMakePair(Qt::Key_A, Qt::NoModifier) << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier)  << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_X, Qt::ControlModifier);
+    keystrokes = {{Qt::Key_T, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_X, Qt::ControlModifier}};
     copyAvailable << true << false << true;
     QTest::newRow(QString("Case7 T,A,A, <- + shift, <- + shift, <- + shift, ctrl + x, undo() | signals: true, false, true").toLatin1())
         << keystrokes << copyAvailable << QString("undo");
@@ -2763,13 +2771,13 @@ namespace {
 
         void drawTextItem(const QPointF &, const QTextItem &textItem) override
         {
-            itemFonts.append(qMakePair(textItem.text(), textItem.font()));
+            itemFonts.append({textItem.text(), textItem.font()});
         }
 
         Type type() const override { return User; }
 
 
-        QList<QPair<QString, QFont> > itemFonts;
+        QList<std::pair<QString, QFont> > itemFonts;
     };
 
     class MyPaintDevice : public QPaintDevice
-- 
2.51.2

